
# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1233868321 18000
# Node ID 4736327a8740579e7c89a8b386de9af0266b07fa
# Parent 7d175e4ea57d21162abbf43a61c5305cfe1995b3
Maildir: Fix to earlier >26 keywords handling change.

--- a/src/lib-storage/index/maildir/maildir-sync-index.c	Wed Feb 04 17:44:01 2009 -0500
+++ b/src/lib-storage/index/maildir/maildir-sync-index.c	Thu Feb 05 16:12:01 2009 -0500
@@ -336,7 +336,7 @@ maildir_sync_mail_keywords(struct maildi
 	   ones. we can get these lists easily by removing common elements
 	   from old and new keywords. */
 	new_indexes = array_get_modifiable(&ctx->keywords, &new_count);
-	for (i = 0; i < old_count && j < new_count; ) {
+	for (i = j = 0; i < old_count && j < new_count; ) {
 		diff = (int)old_indexes[i] - (int)new_indexes[j];
 		if (diff == 0) {
 			array_delete(&ctx->keywords, j, 1);

